### DISCUSSION TITLE
Amazon.CloudWatch.Model.MetricDatum timestamp is not set correctly

### DISCUSSION THREAD

**Discussion by mihneavelinOT** [aws/aws-sdk-net#3333] [Q&A] [answered]
### Describe the bug

Hello,

I am trying to send multiple metrics to CloudWatch using AWSSDK.CloudWatch NuGet via PutMetricDataAsync.
I am sending a List\<MetricDatum\>. When debugging, each MetricDatum has the TimestampUtc attribute set correctly(Timestamp attribute is the same -- I understand from the docs that this is the expected behavior).

When accessing the AWS Console, the datapoints are set wrong, setting the graph time to "UTC timezone" will show me the datapoint as happening with 3 hours before the TimestampUtc I set in the code!!!

### Expected Behavior

To see the datapoints I sent at the correct timestamp in the graph.

### Current Behavior

When accessing the AWS Console, the datapoints are set wrong, setting the graph time to "UTC timezone" will show me the datapoint as happening with 3 hours before the TimestampUtc I set in the code!!!

### Reproduction Steps

Try sending a list of MetricDatum via PutMetricDataAsync function from AWSSDK.CloudWatch NuGet.

### AWS .NET SDK and/or Package version used

AWSSDK.CloudWatch 3.7.304.21

### Targeted .NET Platform

.NET 7

### Operating System and version

Windows 10, Windows 11

**Comment by bhoradc** [comment]
Hi @mihneavelinOT, 

Thank you for reporting this issue. Can you confirm if your code follows the recommendations mentioned in this [blog post](https://aws.amazon.com/blogs/developer/datetime-serialization-changes-in-the-aws-sdk-for-net-and-awspowershell-modules/), for handling UTC timestamp in SDK for .NET?

This Blog post mentions about practices to follow to prevent any potential timezone ambiguity in the code. This would ensure that the time value is treated consistently as UTC, regardless of the local execution environment.

Kindly share a code sample that I can use to reproduce this scenario, in case above recommendations doesn't work or won't apply to your use-case. 

Thank you!

Regards, 
Chaitanya

**Comment by mihneavelinOT** [comment, marked as answer]
Hi @bhoradc,

Thank you so much!!!

I will put my solution here, maybe it will help somebody else.

Before:
When inspecting ```TimestampUtc``` the date time seemed to be correct, but the ```DateTimeKind``` was wrong!!!
```point.timestamp``` is a ```long``` storing the utc timestamp in seconds
```
TimestampUtc = DateTimeOffset.FromUnixTimeSeconds(point.timestamp).DateTime;
```

After:
Notice the explicit specification of the ```DateTimeKind```.
```
var time = DateTimeOffset.FromUnixTimeSeconds(point.timestamp).DateTime;
TimestampUtc = DateTime.SpecifyKind(time, DateTimeKind.Utc);
```

**Comment by bhoradc** [comment]
Hello @mihneavelinOT, 

Glad to know that the blogpost helped. To help others who might be facing similar problems, I will move this Issue to the `Discussions` section. This way, others can find the solution and benefit from the conversation.

Regards, 
Chaitanya
